ledger/eval/txntracer.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package eval
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// TODO do we want something like this?
+type txnGroupDeltas struct {
```

Comment:
Is this supposed to be removed?

Reply:
I've left it there mostly so people could comment about whether or not it makes sense to have some structure containing the identifier as well as the delta. For things such as `/v2/deltas/group/{round}` it will be difficult to correlate the set of returned deltas to which txn group is responsible for those changes.

I can remove it for now and follow up in the next set of PRs.